test/unit/test_basefind.c

```diff
@@ -0,0 +1,134 @@
+}
+
+int test_rz_basefind_with_callbacks(void) {
+	RzBaseFindOpt options;
+	RzList *result = NULL;
+	RzCore *core = rz_core_new();
+	rz_core_file_open_load(core, "bins/firmware/stm32f103-dapboot-v1.20-bluepill.bin", 0, RZ_PERM_R, false);
+
+	// test_basefind_callback_true
+	basefind_options_set_valid(&options);
+	options.callback = test_basefind_callback_true;
+	result = rz_basefind(core, &options);
+	mu_assert_notnull(result, "valid callback (true)");
+	rz_list_free(result);
+
+	// test_basefind_callback_false
+	basefind_options_set_valid(&options);
+	options.callback = test_basefind_callback_false;
+	result = rz_basefind(core, &options);
+	mu_assert_notnull(result, "valid callback (false)");
+	rz_list_free(result);
+
+	rz_core_free(core);
+	mu_end;
+}
+
+int test_rz_basefind_no_callback(void) {
+	RzBaseFindOpt options;
+	RzList *result = NULL;
+	RzCore *core = rz_core_new();
+	rz_core_file_open_load(core, "bins/firmware/stm32f103-dapboot-v1.20-bluepill.bin", 0, RZ_PERM_R, false);
+
+	// valid configuration
+	basefind_options_set_valid(&options);
+	result = rz_basefind(core, &options);
+	mu_assert_notnull(result, "valid pointer_size 32");
+	rz_list_free(result);
+
+	// valid configuration
+	basefind_options_set_valid(&options);
+	options.pointer_size = 64;
+	result = rz_basefind(core, &options);
+	mu_assert_notnull(result, "valid pointer_size 64");
+	rz_list_free(result);
+
+	// pointer_size
+	basefind_options_set_valid(&options);
+	options.pointer_size = 77;
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "invalid pointer_size");
+
+	// min_score
+	basefind_options_set_valid(&options);
+	options.min_score = 0;
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "invalid min_score");
+
+	// min_string_len
+	basefind_options_set_valid(&options);
+	options.min_string_len = 0;
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "invalid min_string_len");
+
+	// alignment
+	basefind_options_set_valid(&options);
+	options.alignment = 0;
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "invalid alignment");
+
+	// start == end
+	basefind_options_set_valid(&options);
+	options.start_address = 0x1111;
+	options.end_address = 0x1111;
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "invalid address (start == end).");
+
+	// start > end
+	basefind_options_set_valid(&options);
+	options.start_address = 0x1111;
+	options.end_address = 0x77;
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "invalid address (start > end).");
+
+	rz_core_free(core);
+	mu_end;
+}
+
+int test_rz_basefind_no_core_load(void) {
+	RzBaseFindOpt options;
+	RzList *result = NULL;
+	RzCore *core = rz_core_new();
+
+	rz_io_open(core->io, "bins/firmware/stm32f103-dapboot-v1.20-bluepill.bin", RZ_PERM_R, 0);
+	basefind_options_set_valid(&options);
+	result = rz_basefind(core, &options);
+	mu_assert_null(result, "file not loaded via core");
+
+	rz_core_free(core);
+	mu_end;
+}
+
+int all_tests() {
```

Comment:
I think these tests should go in `test/integration` because they require bins to be loaded. They also use RzCore so a lot of stuff are just supposed to be in the right place.